Sources/PackageLoading/TargetSourcesBuilder.swift

```diff
@@ -397,6 +400,9 @@ public struct TargetSourcesBuilder {
                 }
             }
 
+            // Ignore README files.
+            if path.basename.hasPrefix("README") { continue}
```

Comment:
I'm not a huge fan of this; I added it because not having it here broke the build (because of a package that was written in C++ but had a `README.md` file in the source directory).

Reply:
Don't think that it is clear cut that readme files in a target are never supposed to be resources and as such ignoring them doesn't seem right. I am curious how these files could have broken the build, unhandled files should only emit warnings?

Reply:
Oh, I see, it is because we're saying `.md` files are now sources, but if they aren't specially formatted, they will break the build?

Reply:
No. It's because there's a package written in C++ that has a `README.md` in its directory, so the package manager code refuses to build it because it thinks it is a mixed-language package.

The literate stuff doesn't care at all about whether or not there is _actually_ code in a Markdown file.

I'm not certain what the right fix is here, though I _am_ certain that this is particular "fix" is a hack and we should do something else.

Reply:
Oh, I see, because `.md` is becoming a source file type, it is seen as mixed source. 

Would it make sense to have a more specific file extension for the literate programming stuff, so that we can distinguish them from "regular" markdown files? In principle, it could be correct that it is mixed source if the .md file *does* contain code.

Another approach could be only treating .md files as source if they're part of a target that isn't a C-family one, but that seems awfully specific.

Reply:
That would make sense to me.  Using plain `.md` as files to be compiled seems that it would surprise a lot of users.